File: ios/HMP/HMP/Features/Common/Design/AnimationTokens.swift
import SwiftUI

/// 动画令牌 - 与 Android AnimationTokens.kt 保持对等
struct AnimationTokens {
    // MARK: - 持续时间 (秒)
    static let microInteraction: Double = 0.2     // 微交互动画
    static let transition: Double = 0.4           // 过渡动画
    static let complex: Double = 0.65             // 复杂动画
    static let background: Double = 3.0           // 背景动画

    // MARK: - 缓动 (UnitCurve 对应 CubicBezierEasing)
    static let easeInOut = UnitCurve.easeInOut
    static let easeOut = UnitCurve.easeOut
    static let easeIn = UnitCurve.easeIn

    // MARK: - 弹簧动画
    static func springMedium() -> Animation {
        .interactiveSpring(response: 0.3, dampingFraction: 0.7, blendDuration: 0.3)
    }

    static func springBouncy() -> Animation {
        .interactiveSpring(response: 0.4, dampingFraction: 0.5, blendDuration: 0.3)
    }

    static func springGentle() -> Animation {
        .interactiveSpring(response: 0.5, dampingFraction: 0.6, blendDuration: 0.3)
    }
}

File: desktop/feature-ui/src/desktopMain/kotlin/com/hmp/desktop/ui/design/AnimationTokens.kt
package com.hmp.desktop.ui.design

import androidx.compose.animation.core.CubicBezierEasing
import androidx.compose.animation.core.Spring
import androidx.compose.animation.core.spring

object AnimationTokens {
    const val MICRO_INTERACTION = 200
    const val TRANSITION = 400
    const val COMPLEX = 650
    const val BACKGROUND = 3000

    val EASE_IN_OUT = CubicBezierEasing(0.4f, 0.0f, 0.2f, 1.0f)
    val EASE_OUT = CubicBezierEasing(0.2f, 0.0f, 0.1f, 1.0f)
    val EASE_IN = CubicBezierEasing(0.6f, 0.0f, 0.8f, 1.0f)

    val SPRING_MEDIUM = spring<Float>(
        stiffness = Spring.StiffnessMediumLow,
        dampingRatio = Spring.DampingRatioMediumBouncy
    )

    val SPRING_BOUNCY = spring<Float>(
        stiffness = Spring.StiffnessMedium,
        dampingRatio = Spring.DampingRatioHighBouncy
    )

    val SPRING_GENTLE = spring<Float>(
        stiffness = Spring.StiffnessLow,
        dampingRatio = Spring.DampingRatioHighBouncy
    )
}

enum class AnimationDirection {
    LEFT, RIGHT, TOP, BOTTOM, CENTER
}
